pkg/server/ca/manager.go

```diff
@@ -811,6 +818,40 @@ func (m *Manager) notify(ctx context.Context, event string, advise bool, pre fun
 	return nil
 }
 
+func (m *Manager) filterInvalidEntries(ctx context.Context, entries *journal.Entries) ([]*JWTKeyEntry, []*X509CAEntry, error) {
+	bundle, err := m.fetchOptionalBundle(ctx)
+
+	if err != nil {
+		return nil, nil, err
+	}
+
+	filteredEntriesJwtKeys := []*JWTKeyEntry{}
+	filteredEntriesX509CAs := []*X509CAEntry{}
+
+	// certificate append to the database whenever upstreamClient does not exist
+	if m.upstreamClient != nil {
+		filteredEntriesX509CAs = entries.X509CAs
+	} else {
+		for _, entry := range entries.GetX509CAs() {
+			if containsX509CA(bundle.RootCas, entry.Certificate) {
+				filteredEntriesX509CAs = append(filteredEntriesX509CAs, entry)
+			} else {
+				m.c.Log.Warn("Dropping X509 CA as it not found in bundle")
```

Comment:
Since this will no doubt be a very strange situation, I think being as descriptive as we possibly can here is going to be important

```suggestion
				m.c.Log.Warn("Root CA certificate found in journal on disk but not present in bundle from datastore; SPIRE will now rotate to recover.")
```

Same comment for the JWT key log message